woocommerce/woocommerce-ios
# REST API: Handle edge cases for Jetpack setup

Closes #8919


## Description


This PR fixes 2 edge cases for Jetpack setup discussed in [pe5sF9-1oh-p2]:
- When a user sets up WCPay for there non-Jetpack site, tapping on the install button on the Jetpack benefit modal will authenticate them with the connected WPCom account and show the install flow for JCP right after the sync is done. Please note: the install flow for JCP doesn't work currently due to the outdated code in JCP, so please ignore the issue until this is fixed on the backend. (Internal discussion: p1680512610989549-slack-CBG1CP4EN)
- When a site has a custom URL for its admin page, Jetpack connection will fail with error code 404. The fix is to detect the error and show the error screen with a suggestion to connect on a browser or contact support.

Also: the feature flag for Jetpack setup is enabled to release this feature for 13.1.

## Testing instructions

1. Testing JCP install
- Prerequisite: make sure that you have a self-hosted Woo store without Jetpack.
- Log in to the test store if needed.
- Install, activate, and connect WCP to your WPCom account.
- Navigate to the Jetpack setup flow by tapping the Jetpack benefit banner on the dashboard screen or go to Menu > Settings > Install Jetpack.
- Log in to your WPCom account as part of the Jetpack setup flow. After the login succeeds, a loading screen should appear, and then you should be authenticated with the account.
- Notice that the setup flow for JCP will then be presented.

2. Testing custom admin page.
- Make sure that WCPay is deactivated on your test store.
- Install and activate [WP Hide](https://wordpress.org/plugins/wp-hide-security-enhancer/) plugin on your store.
- On wp-admin left side bar navigate to WP Hide > Hide Admin. Update a custom URL and enable "Block default Admin Url"
<img width="1087" alt="Screenshot 2023-04-04 at 09 35 33" src="https://user-images.githubusercontent.com/5533851/229679826-54444b80-d040-4eef-a4cd-7d8fbf9dfe97.png">

- On the app, navigate to the Jetpack setup flow by tapping the Jetpack benefit banner on the dashboard screen or go to Menu > Settings > Install Jetpack.
- During the connection step of the setup flow, notice that the connection web view is dismissed and the setup screen should show an error about the connection.

## Screenshots



https://user-images.githubusercontent.com/5533851/229794970-11eb2627-d95c-4720-8229-32bcb7cea40e.mp4





---
